hacking/shippable/download.py

```diff
@@ -205,181 +145,34 @@ def download_run(args):
-            download(args, headers, path, url, is_json=False)
-
-        if args.test_results:
-            path = os.path.join(output_dir, '%s/test.json' % job_number)
-            url = 'https://api.shippable.com/jobs/%s/jobTestReports' % job_id
-            download(args, headers, path, url)
-            extract_contents(args, path, os.path.join(output_dir, '%s/test' % job_number))
-
-        if args.coverage_results:
-            path = os.path.join(output_dir, '%s/coverage.json' % job_number)
-            url = 'https://api.shippable.com/jobs/%s/jobCoverageReports' % job_id
-            download(args, headers, path, url)
-            extract_contents(args, path, os.path.join(output_dir, '%s/coverage' % job_number))
-
-
-def extract_contents(args, path, output_dir):
-    """
-    :type args: any
-    :type path: str
-    :type output_dir: str
-    """
-    if not args.test:
-        if not os.path.exists(path):
-            return
-
-        with open(path, 'r') as json_fd:
-            items = json.load(json_fd)
-
-            for item in items:
-                contents = item['contents'].encode('utf-8')
-                path = output_dir + '/' + re.sub('^/*', '', item['path'])
-
-                directory = os.path.dirname(path)
-
-                if not os.path.exists(directory):
-                    os.makedirs(directory)
-
-                if args.verbose:
-                    print(path)
-
-                if path.endswith('.json'):
-                    contents = json.dumps(json.loads(contents), sort_keys=True, indent=4).encode('utf-8')
-
-                if not os.path.exists(path):
-                    with open(path, 'wb') as output_fd:
-                        output_fd.write(contents)
-
-
-def download(args, headers, path, url, is_json=True):
-    """
-    :type args: any
-    :type headers: dict[str, str]
-    :type path: str
-    :type url: str
-    :type is_json: bool
-    """
-    if args.verbose or args.test:
-        print(path)
-
-    if os.path.exists(path):
-        return
-
-    if not args.test:
-        response = requests.get(url, headers=headers)
-
-        if response.status_code != 200:
-            path += '.error'
-
-        if is_json:
-            content = json.dumps(response.json(), sort_keys=True, indent=4).encode(response.encoding)
-        else:
-            content = response.content
-
-        directory = os.path.dirname(path)
-
-        if not os.path.exists(directory):
-            os.makedirs(directory)
-
-        with open(path, 'wb') as content_fd:
-            content_fd.write(content)
-
-
-def get_api_key():
-    """
-    rtype: str
-    """
-    key = os.environ.get('SHIPPABLE_KEY', None)
-
-    if key:
-        return key
-
-    path = os.path.join(os.environ['HOME'], '.shippable.key')
-
-    try:
-        with open(path, 'r') as key_fd:
-            return key_fd.read().strip()
-    except IOError:
-        return None
+    if args.coverage_results:
```

Comment:
I'd rename this and the associated CLI option to `artifacts`, since not all artifacts are coverage artifacts.